oceanprefilter/utils.go

```diff
@@ -56,76 +56,24 @@ func toGray(pic image.Image) *image.Gray {
 	return gray
 }
 
-// crop the image from yValue -> img.Bounds().Max.Y
-// and then split the cropped image into nh horizontal and nv vertical bands of equal height and width
-func splitUpImage(img image.Image, exZone *image.Rectangle, yValue, nh, nv int) ([]image.Image, error) {
-	if img == nil {
-		return nil, errors.New("input image to split up is nil")
-	}
-	if nv <= 0 {
-		return nil, fmt.Errorf("number of vertical lines must be greater than 0")
-	}
-	if nh <= 0 {
-		return nil, fmt.Errorf("number of horizontal lines must be greater than 0")
-	}
-
-	// Crop the image from yValue to img.Bounds().Max.Y
-	bounds := img.Bounds()
-	croppedHeight := bounds.Max.Y - yValue
-	if croppedHeight <= 0 {
-		return nil, fmt.Errorf("yValue must be within the image bounds")
-	}
-	// edit exluded zone to take the crop into account
-	excludedBox := image.Rectangle{}
-	if exZone != nil {
-		excludedBox.Min.X = exZone.Min.X
-		excludedBox.Min.Y = exZone.Min.Y - yValue
-		excludedBox.Max.X = exZone.Max.X
-		excludedBox.Max.Y = exZone.Max.Y - yValue
-	}
-	croppedRect := image.Rect(bounds.Min.X, yValue, bounds.Max.X, bounds.Max.Y)
-	croppedImg := image.NewRGBA(croppedRect)
-	draw.Draw(croppedImg, croppedImg.Bounds(), img, croppedRect.Min, draw.Src)
-
-	// Split the cropped image into n horizontal bands
-	bandHeight := croppedImg.Bounds().Dy() /nh
-	bandWidth := croppedImg.Bounds().Dx() / nv
-	images := make([]image.Image, 0, nv*nh)
-	for i := 0; i < nh; i++ {
-		for j := 0; j < nv; j++ {
-			bandRect := image.Rect(j*bandWidth, i*bandHeight, (j+1)*bandWidth, (i+1)*bandHeight)
-			// if rect in excluded zone, skip it
-			if exZone != nil && bandRect.Overlaps(excludedBox) {
-				continue
-			}
-			bandImg := image.NewRGBA(bandRect)
-			draw.Draw(bandImg, bandImg.Bounds(), croppedImg, image.Point{bandRect.Min.X, bandRect.Min.Y + yValue}, draw.Src)
-
-			images = append(images, bandImg)
-		}
-	}
-	return images, nil
-}
-
-
 // crop the image from yValue -> img.Bounds().Max.Y
 // and then split the cropped image into nh horizontal and nv vertical bands of equal height and width (dimensions given)
 func splitUpImageConst(img image.Image, exZone *image.Rectangle, yValue, h, w int) ([]image.Image, error) {
 	if img == nil {
 		return nil, errors.New("input image to split up is nil")
 	}
 	if h <= 0 {
-		return nil, fmt.Errorf("height must be greater than 0")
+		return nil, fmt.Errorf("height must be greater than 0, got %v", h)
```

Comment:
rather than using the `fmt` package, use the `errors` package, which has the same Funcion